src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -273,6 +271,10 @@ protected internal bool CheckWitnessInternal(UInt160 hash)
                 }
                 return false;
             }
+            else
```

Comment:
This code looks so weird

Reply:
It reduces one condition in the positive case (the usual one)